agent/agent.go

```diff
@@ -1187,27 +1187,29 @@ func (a *Agent) shouldKill(tc *taskContext, ignoreTaskGroupCheck bool) bool {
 
 // logPanic logs a panic to the task log and returns the panic error, along with
 // the original error (if any). If there was no panic error, this is a no-op.
-func (a *Agent) logPanic(logger client.LoggerProducer, pErr, originalErr error, op string) error {
+func (a *Agent) logPanic(tc *taskContext, pErr, originalErr error, op string) error {
 	if pErr == nil {
 		return nil
 	}
 
 	catcher := grip.NewBasicCatcher()
 	catcher.Add(originalErr)
 	catcher.Add(pErr)
-	if logger != nil && !logger.Closed() {
-		logMsg := message.Fields{
-			"message":   "programmatic error: Evergreen agent hit panic",
-			"operation": op,
-		}
+	logMsg := message.Fields{
+		"message":   "programmatic error: Evergreen agent hit panic",
+		"operation": op,
+	}
+	if tc.logger != nil && !tc.logger.Closed() {
 		if a.opts.HostID != "" {
 			logMsg["host_id"] = a.opts.HostID
 		}
 		if a.opts.PodID != "" {
 			logMsg["pod_id"] = a.opts.PodID
 		}
-		logger.Task().Error(logMsg)
+		tc.logger.Task().Error(logMsg)
 	}
+	logMsg["task_id"] = tc.task.ID
```

Comment:
Can we also include the task execution? Might help narrow down what to look at a little faster.